### PR TITLE
make a basic User model (including secure passwords)

### PR DESCRIPTION
**Pull request by minhthuc**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,3 +1,8 @@
 class User < ActiveRecord::Base
-    has_many :microposts
+    validates :name, presence: true, length: {maximum: 50}
```
Comment (duytd) [inline app/models/user.rb]: [IMO] We should use constant for `5`

```ruby
MAX_NAME_LENGTH = 5
```

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
+require 'spec_helper'
+
+describe User do
+  before do 
+    @user = User.new(name: "Example User", email: "user@example.com",
```
[the remark this answers is not in the document]
Comment (minhthuc) [reply spec/models/user_spec.rb]: i think this gem will create a valid model base on model validation that can reuse. However, i have not used.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  subject { @user }
+
+  it { should respond_to(:name) }
+  it { should respond_to(:email) }
+  it { should respond_to(:password_digest) }
+  it { should respond_to(:password) }
+  it { should respond_to(:password_confirmation) }
+
+  context "when name is not present" do
+    before {@user.name=" "};
```
Comment (duytd) [inline spec/models/user_spec.rb]: [IMO] We should use `nil` (name is not present)

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                     foo@bar_baz.com foo@bar+baz.com]
+      addresses.each do |invalid_address|
+        @user.email = invalid_address
+        expect(@user).not_to be_valid
+      end
+    end
+  end
+
+  context "when password is not present" do
+    before do
```
Comment (duytd) [inline spec/models/user_spec.rb]: [IMO] We should use `nil` (password is not present)

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  context "when email format is valid" do
+    it "should be valid" do
+      addresses = %w[user@foo.COM A_US-ER@f.b.org frst.lst@foo.jp a+b@baz.cn]
+      addresses.each do |valid_address|
+        @user.email = valid_address
+        expect(@user).to be_valid
+      end
+    end
+  end
+  context "when email address is already taken" do
```
Comment (duytd) [inline spec/models/user_spec.rb]: [IMO] Missing new line above

**Comment by duytd** [comment]
LGTM

**Pull request merged**